packages/react-core/src/components/Modal/ModalContent.tsx

```diff
@@ -72,8 +72,12 @@ export const ModalContent: React.FunctionComponent<ModalContentProps> = ({
 
   const ariaLabelledbyFormatted = (): string => {
     const idRefList: string[] = [];
-    if (ariaLabel && boxId) {
-      idRefList.push(ariaLabel && boxId);
+    if (ariaLabel) {
+      // we shouldn't set an aria-labelledby attribute if we have an aria-label
+      return undefined;
+    }
+    if (boxId) {
+      idRefList.push(boxId);
     }
     if (ariaLabelledby) {
       idRefList.push(ariaLabelledby);
```

Comment:
We may want to only allow either boxId or aria-labelledby to be pushed to idRefList, not both. Right now the `pf-v6-c-modal-box` wrapper element may have extraneous labeling if an aria-labelledby is passed and that value references an element inside the Modal (like the [basic modal example](https://patternfly-react-pr-11012.surge.sh/components/modal#basic-modals) is doing).

So right now in that basic example, the accessible name of the `dialog` is *everything* in it, followed by its title (which is already announced). Having the `boxId` act as a fallback if both aria-label and aria-labelledby would work better. Another thing, though, is that the check we're doing in the Modal code (~line 135) isn't really necessary, because if a consumer doesn't pass an ID to the Modal then we're constructing it, so we're always applying an aria-labelledby.

Unless I'm missing a case where that `boxId` would actually never be a valid value and aria-labelledby wouldn't automatically be applied or anything.